ckanext/datapusher/logic/action.py

```diff
@@ -84,18 +84,33 @@ def datapusher_submit(context, data_dict):
         'error': '{}',
     }
     try:
-        task_id = p.toolkit.get_action('task_status_show')(context, {
+        existing_task = p.toolkit.get_action('task_status_show')(context, {
             'entity_id': res_id,
             'task_type': 'datapusher',
             'key': 'datapusher'
-        })['id']
-        task['id'] = task_id
+        })
+
+        if existing_task and existing_task.get('state') == 'pending':
```

Comment:
if task does not exist, [error is raised](https://github.com/ckan/ckan/blob/42406b15de78bff6e486a0e263f98114c644fffc/ckan/logic/action/get.py#L2348). In that case, could first check be omitted in order to avoid confusing expectations in future?